### PR TITLE
acquisitions: export accounts

### PR DESCRIPTION
**Pull request by lauren-d**
* Adds config to accounts route.
* Configures export button component.

Co-Authored-by: Lauren-D <laurent.dubois@itld-solutions.be>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on `rero-ils#<xx>`'s PR(s):

* rero/rero-ils#2957
* rero/ng-core#504

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts [hunk 1]
```diff
@@ -31,30 +34,65 @@ export class AccountListComponent implements OnInit {
   /** Root account to display */
   rootAccounts: IAcqAccount[] = [];
 
+  /**
+   * Export options configuration.
+   */
```
Comment (zannkukai) [inline projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:39]: please use a single line comment to describe a component attribute.

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts [hunk 2]
```diff
@@ -63,4 +101,57 @@ export class AccountListComponent implements OnInit {
...
+          return {
+            label: format.label,
+            url: this.getExportFormatUrl(format),
+            disabled: !this.canExport(format),
+            disabled_message: this.exportInfoMessage
+          };
+        }
+      );
+    }
+    return [];
```
Comment (zannkukai) [inline projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:121]: reverting the `if` condition will saves some indentation spaces.
```javascript
if (exportFormat === undefined || exportFormat.length === 0) {
    return [];
}
```
Maybe using a default "empty list" value on `exportFormats` attribute will save you from this condition.

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts [hunk 3]
```diff
@@ -63,4 +101,57 @@ export class AccountListComponent implements OnInit {
...
+    ];
+    const query = defaultQueryParams.join(' AND ')
+    const baseUrl = format.endpoint
+      ? format.endpoint
+      : this._apiService.getEndpointByType('acq_accounts');
+    let url = `${baseUrl}?q=${encodeURIComponent(query)}&format=${format.format}`;
+    // check if max rest result size is disabled
+    if (!format.disableMaxRestResultsSize) {
+      url += `&size=${RecordService.MAX_REST_RESULTS_SIZE}`;
+    }
```
Comment (zannkukai) [inline projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:142]: I should be prettier to use `HttpParams` to build a list of queryParams and build the query url params based ton this object. I'm not fan creating URL by hands.

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts (hunk 1 above)
Comment (lauren-d) [reply projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts]: done

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts (hunk 2 above)
Comment (lauren-d) [reply projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts]: done

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts (hunk 3 above)
Comment (lauren-d) [reply projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts]: ok

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.html [hunk 4]
```diff
@@ -18,18 +18,28 @@
 <ng-container *ngIf="organisation">
   <header class="row d-flex align-items-baseline">
     <h1 class="col">{{ 'Acquisition accounts' | translate }}</h1>
-    <div class="col-3 text-right">
-      <a [routerLink]="['/', 'acquisition', 'accounts', 'transfer']"
-         class="btn btn-sm btn-outline-primary mr-2"
-      >
-        <i class="fa fa-exchange"></i> {{ 'Transfer funds' | translate }} &hellip;
-      </a>
+    <div class="col text-right">
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply projects/admin/src/app/acquisition/components/account/account-list/account-list.component.html:21]: According doc from bootstrap 4, we can use only `col` for equal width.
https://getbootstrap.com/docs/4.0/layout/grid/#:~:text=sm%2D6.-,Equal%2Dwidth,-For%20example%2C%20here

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts [hunk 5]
```diff
@@ -20,6 +20,10 @@ import { UserService } from '@rero/shared';
 import { OrganisationService } from '../../../../service/organisation.service';
 import { IAcqAccount } from '../../../classes/account';
 import { AcqAccountApiService } from '../../../api/acq-account-api.service';
+import { ApiService, RecordService } from '@rero/ng-core';
+import { TranslateService } from '@ngx-translate/core';
+import { exportFormats } from '../../../routes/accounts-route';
+import {HttpParams} from '@angular/common/http';
```
Comment (zannkukai) [inline projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:26]: spaces between brackets ;-)

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts [hunk 6]
```diff
@@ -31,30 +35,61 @@ export class AccountListComponent implements OnInit {
...
+
 
   // GETTER & SETTER ============================================================
   /** Get the current organisation */
   get organisation(): any {
     return this._organisationService.organisation;
   }
 
+  /**
+   * Return a message containing the reasons why record list cannot be exported
```
Comment (zannkukai) [inline projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:57]: `Get a message ...`

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts [hunk 7]
```diff
@@ -31,30 +35,61 @@ export class AccountListComponent implements OnInit {
...
+   * @return Boolean
```
Comment (zannkukai) [inline projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:58]: Bad return statement ; should be 'string'. In general, for getter functions, we skip the return docstring line

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts (hunk 6 above)
Comment (lauren-d) [reply projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:57]: done

File: projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts (hunk 7 above)
Comment (lauren-d) [reply projects/admin/src/app/acquisition/components/account/account-list/account-list.component.ts:58]: ok

**Comment by Garfield-fr** [comment]
Why you are 7 commits ? Please rebase

**Pull request merged**